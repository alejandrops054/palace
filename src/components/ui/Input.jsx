import React from 'react';
import {Link} from 'react-router-dom'
//icon: clase de Font-awesome
//type: text | number | password
//label: string
//placeholder: string
//col: columna de bootstrap

export default function Input ({icon, type='', label='', placeholder='', col='', onChange, name, value, disabled, onClick, url}){
  return (
    <div className={`col-lg-${ col } mb-3`}>
      <label>{ label }</label>
      <div className="input-group">
        <input onChange={ onChange } type={ type } placeholder={ placeholder } className='form-control' name={ name } value={ value } disabled = {disabled}/>
        {icon && url ?(<Link to={url}><button  onClick={onClick} className="input-group-addon btn btn-secondary" type="button" style={{borderTopRightRadius:'.3rem', borderBottomRightRadius:'.3rem'}}><i className={`${ icon }`}></i></button></Link> ) :null }
        {icon && !url?(<button  onClick={onClick} className="input-group-addon btn btn-secondary" type="button" style={{borderTopRightRadius:'.3rem', borderBottomRightRadius:'.3rem'}}><i className={`${ icon }`}></i></button>) :null }
      </div>
    </div>
  );
}




