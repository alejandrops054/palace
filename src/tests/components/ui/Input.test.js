import React from 'react'
import { shallow } from 'enzyme'

import Input from '../../../components/ui/Input'


describe('Prueba componente UI Input', () => {
    test('Debe de mostrar el componente <Input/>', () => {
        const store = {
            icon: '',
            type: '',
            label: '',
            placeholder: '',
            col: ''
        }
        const wrapper = shallow(<Input icon={store.icon} type={store.type} label={store.label} placeholder={store.placeholder} col={store.col} />);

        expect(wrapper).toMatchSnapshot();
    })

})