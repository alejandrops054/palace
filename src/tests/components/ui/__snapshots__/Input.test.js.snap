// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba componente UI Input Debe de mostrar el componente <Input/> 1`] = `
<div
  className="col-lg- mb-3"
>
  <label />
  <div
    className="input-group"
  >
    <input
      className="form-control"
      placeholder=""
      type=""
    />
  </div>
</div>
`;
